### PR TITLE
node compatibility

### PR DESCRIPTION
**Pull request by cirospaciari**
This modifications are openssl/boringssl only

- [x] add cert option
- [x] add key option
- [x] add ca list option
- [x] add cert list option
- [x] add key list option
- [x] request_cert
- [x] reject_unauthorized
- [x] handshake callback and error

Also fixes: https://github.com/oven-sh/bun/issues/1263

### CODE REVIEW

File: src/crypto/openssl.c [hunk 1]
```diff
@@ -534,9 +643,398 @@ SSL_CTX *create_ssl_context_from_options(struct us_socket_context_options_t opti
...
+  }
+  ret = SSL_CTX_use_PrivateKey(ctx, pkey);
+  EVP_PKEY_free(pkey);
+
+end:
+  BIO_free(in);
+  return ret;
+}
+
+X509 * SSL_CTX_get_X509_from(SSL_CTX *ctx, const char *content) {
```
Comment (Jarred-Sumner) [inline src/crypto/openssl.c]: are these functions not included in boringssl? they read like boringssl functions

Comment (cirospaciari) [reply src/crypto/openssl.c]: Is not, these are custom, I think is better to rename this to be more like uSockets name looking

File: src/crypto/openssl.c [hunk 2]
```diff
@@ -534,9 +672,398 @@ SSL_CTX *create_ssl_context_from_options(struct us_socket_context_options_t opti
...
+        const long set_tmp_dh = SSL_CTX_set_tmp_dh(ssl_context, dh_2048);
+        DH_free(dh_2048);
+
+        if (set_tmp_dh != 1) {
+            free_ssl_context(ssl_context);
+            return NULL;
+        }
+
+        /* OWASP Cipher String 'A+' (https://www.owasp.org/index.php/TLS_Cipher_String_Cheat_Sheet) */
+        if (SSL_CTX_set_cipher_list(ssl_context, "DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256") != 1) {
```
Comment (Jarred-Sumner) [inline src/crypto/openssl.c:1044]: is this supposed to go in this scope? or is it supposed to be set when `options.SSL_ciphers` is not set

Comment (cirospaciari) [reply src/crypto/openssl.c:1044]: Actually that's no need to set this cyphers manually at this stage. I will update this.

Comment (cirospaciari) [reply src/crypto/openssl.c:1044]: Taking a deeper look is actually fine, will be only the default if no ciphers are informed and only when dh_params_file_name is informed, we can revisit it when we add dh_params (in memory) support.

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**